lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -727,72 +837,154 @@ def _labeled_unit_config(
         return static_config
 
 
+def _resolve_dir_against_main_path(*path_elements: str) -> Optional[str]:
```

Comment:
Why does this have to be and `Optional[str]` return value. An empty string is false in any boolean context. So why not always return a string and in the failed case return and empty string. It is generally a good idea to return the same type from the same method.